mmf/datasets/processors/processors.py

```diff
@@ -1493,3 +1493,29 @@ def __call__(self, item: Union[Dict[str, Any], str]) -> Dict[str, Any]:
         assert class_index != -1, f"{label} is not present in vocab file"
 
         return {"class_index": torch.tensor(class_index, dtype=torch.long)}
+
+
+@registry.register_processor("detr_image")
+class DETRImageProcessor(BaseProcessor):
+    def __init__(self, config, *args, **kwargs):
+        super().__init__(config, *args, **kwargs)
+
+        self.image_size = config.image_size
+        self.max_size = config.max_size
+
+        import mmf.modules.detr.datasets.transforms as T
+
+        self.detr_im_transform = T.Compose(
+            [
+                T.RandomResize([config.image_size], max_size=self.max_size),
+                T.ToTensor(),
+                T.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225]),
+            ]
+        )
```

Comment:
We should also be able to remove the `detr_image` specific modifications to the datasets in the above files.

Reply:
(this is moved to the dataset PR: https://github.com/facebookresearch/mmf/pull/740)

Since `RandomResize` is not available in torchvision_transforms, I kept the DETRImageProcessor there.